### ISSUE TITLE
Implement [pdffile]_updates.plist in pdf parser

### ISSUE THREAD

**Issue by libreliodev** [libreliodev/android#300]
This is related to https://github.com/libreliodev/iOS/issues/141

**Comment by intrications** [comment]
@libreliodev 

As was a problem with the iOS version, I don't see anywhere that exposes the obj ID.

Was it decided to use "IndexAtPage" instead?

**Comment by libreliodev** [comment]
@intrications 

> Was it decided to use "IndexAtPage" instead?

Yes

**Comment by intrications** [comment]
@libreliodev 

Does "p1" refer to page 1? (Which possible is referred to as page 0 when using a zero based array in the code.

**Comment by libreliodev** [comment]
@serverfire Can you answer this question:

> Does "p1" refer to page 1? (Which possible is referred to as page 0 when using a zero based array in the code.

**Comment by serverfire** [comment]
@libreliodev 
> Does "p1" refer to page 1? (Which possible is referred to as page 0 when using a zero based array in the code.

Yes. Iteration starts at 1 to n.

**Comment by libreliodev** [comment]
@intrications In case you missed it, yes, iteration starts at 1 to n. Thanks @serverfire

**Comment by intrications** [comment]
@libreliodev 

I'm able to download wind_384_updates.plist using the com.developer.wind server.

This is meant to be the updates file of Wind 384 Sample, I think.

However, I don't understand where the clickable areas are meant to be in the magazine.

Does "p1" refer to the cover page on this issue? Clicking anywhere on it doesn't seem to do anything.

**Comment by serverfire** [comment]
@intrications @libreliodev 
> Does "p1" refer to the cover page on this issue? 

Yes. We checked it before. It was working. 
Specified `rect` in plist file is in the same as pdf coordinate system. rect is taken from `pdf.js` output which is exact value in pdf file.

**Comment by serverfire** [comment]
Last comment has updated.

**Comment by intrications** [comment]
Which part of the cover should I click on to get a response? I want to make sure it is working correctly before I try to process the updates file. There is no point trying to remove a link that isn't even working.

**Comment by intrications** [comment]
It seems to me that only the paid version of the issue has these links.

**Comment by libreliodev** [comment]
@intrications You may want to use the javascript console to test links.

Go to https://admin.librelio.com, enter the credentials given to you by private message, and enter `developer`as folder name.

The select application `Wind`in upper menu, and then select `wind_384`in `publications`, then `edit links`

**Comment by intrications** [comment]
@libreliodev 

That helps.

Is "Edit" a possible action in the plist? I can only create "Add" and "Remove".

**Comment by intrications** [comment]
The plist is no longer giving me "IndexAtPage", e.g.:

```
<dict>
  <key>p1</key>
  <array>
    <dict>
      <key>Action</key>
      <string>Remove</string>
      <key>ID</key>
      <integer>806</integer>
    </dict>
    <dict>
      <key>Action</key>
      <string>Add</string>
      <key>ID</key>
      <integer>807</integer>
      <key>Rect</key>
      <string>41.22120715294429 554.5056321471409 240.51727939964036 795.8057244548332</string>
      <key>Link</key>
      <string>goto://6</string>
    </dict>
  </array>
</dict>
</plist>
```

**Comment by libreliodev** [comment]
@intrications No, I don't think there is a `Edit`action in the plist. `Edit` = `Remove` + `Add`

**Comment by intrications** [comment]
@libreliodev Thanks for confirming. I'm trying to piece together the information from the various github issues and I saw an Edit action somewhere - can't find it anymore. But I did see that the plist now used Remove + Add to achieve the same.
